File: Trees/BinaryTreeJava/src/binarytree/LeftView.java
package binarytree;

import java.util.ArrayList;

public class LeftView {
    public static void main(String[] args) {
        Node root = null;
        BinaryTree s = new BinaryTree();
        root = s.buildTree(root);
        ArrayList<Integer> ans  = leftView(root);
        System.out.println(ans);
        //1 2 4 -1 -1 5 -1 -1 3 6 -1 -1 -1
    }
    private static void solve(Node root , ArrayList<Integer> ans , int level){
        if(root == null){
            return;
        }

        if(level == ans.size()){
            ans.add(root.data);
        }

        solve(root.left , ans , level + 1);
        solve(root.right , ans , level+1);
    }

    public static ArrayList<Integer> leftView(Node root) {
        ArrayList<Integer> ans  = new ArrayList<>();
        solve(root,ans,0);
        return ans ;
    }
}

File: Trees/BinaryTreesC++/leftView.cpp
#include <bits/stdc++.h>
using namespace std;

class Node{
	public:
		int data;
		Node* left;
		Node* right;
		Node(int d){
			this->data = d;
			this->left = NULL;
			this->right = NULL;
		}
};
Node* buildTree(Node* root){	
	int data; 
	cin>>data ;
	root = new Node(data);
	if(data == -1){
		return NULL;
	}
	root->left = buildTree(root->left);
	root->right = buildTree(root->right);
	return root;
}

void solve(Node*root , vector<int> &ans ,int level){
	if(root == NULL){
		return ;
	}

	if(level == ans.size()){
		ans.push_back(root->data);
	}

	solve(root->left , ans , level+1);
	solve(root->right , ans , level+1);

}

vector<int> leftView(Node* root){
	vector<int> ans;
	int level = 0;
	solve(root , ans , level);
	return ans;
}

int main(){
	Node* root = NULL ;
	root = buildTree(root);
	vector<int> ans  = leftView(root);
	for(auto i : ans){
		cout<< i <<" ";
	}
}
